src/components/coupon/CouponCheckItem.tsx

```diff
@@ -0,0 +1,62 @@
+import styled from "@emotion/styled";
+import CheckboxLabel from "../inputs/CheckboxLabel";
+import Flex from "../styled/Flex";
+import LabelTextPair, { labelTextPairType } from "./LabelTextPair";
+import { useCouponContext } from "../../pages/order-confirm/context/CouponProvider";
+import { Coupon } from "../../api/coupon";
+
+interface CouponCheckItemProps {
+  coupon: Coupon;
+  details?: labelTextPairType[];
+  disabled?: boolean;
+}
+
+function CouponCheckItem({
+  coupon,
+  details,
+  disabled = false,
+}: CouponCheckItemProps) {
+  const { selectedCoupons, handleCouponToggle } = useCouponContext();
+  const [year, month, day] = coupon.expirationDate;
+
+  return (
+    <Container
+      justifyContent="flex-start"
+      alignItems="flex-start"
+      gap="sm"
+      as={"li"}
+      disabled={disabled}
+    >
+      <CheckboxLabel
+        isChecked={selectedCoupons.some(
+          (selectedCoupon) => selectedCoupon.id === coupon.id
+        )}
```

Comment:
selectedCoupons를 Set으로 관리하면
```suggestion
        isChecked={selectedCoupons.has(coupon.id)}
```
이렇게도 사용할 수 있을거 같은데 어떻게 생각해??